components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -938,6 +975,22 @@ public static void setCookie(HttpServletRequest req, HttpServletResponse resp, S
         setCookie(req, resp, cookieName, id, age, setSameSite, null);
     }
 
+    /**
+     * return the restart login flow status.
+     * @return status of the restart login flow.
+     */
+    public static Boolean isRestartLoginFlow() {
+        return restartLoginFlow;
+    }
+
+    /**
+     * Enable or disable the restart login flow.
+     * @param status status of the restart login flow.
+     */
+    public static void setRestartLoginFlow(Boolean status) {
+        restartLoginFlow = status;
+    }
```

Comment:
we can't keep the restartLoginFlow flag as a FrameworkUtil variable. Think of concurrent flows.